src/gwt/src/org/rstudio/core/client/VirtualConsole.java

```diff
@@ -698,6 +761,20 @@ public String debugDump()
       public final SpanElement element;
    }
 
+   interface Resources extends ClientBundle
+   {
+      @Source("./virtualscroller/virtualscroller.js")
+      TextResource virtualScrollerCode();
+   }
+
+   private static final Resources RES = GWT.create(Resources.class);
+
+   private static final ExternalJavaScriptLoader virtualScrollerLoader_ =
+      new ExternalJavaScriptLoader(VirtualScrollerResources.INSTANCE.virtualscrollerjs().getSafeUri().asString());
+   private static VirtualScrollerNative virtualScrollerNative_;
```

Comment:
Maybe I'm misreading this, but wouldn't we need one instance of `VirtualScrollerNative` per `VirtualConsole`?

Reply:
Addressed in the commit I just pushed. Unfortunately just making it an instance variable wasn't an option as the way that VirtualConsole works is frankly kind of broken. A new instance is created pretty much any time you hit enter and there's a lot of cases where it's created with a `null` parent. I tried to decouple it from Gin but I wasn't able to make a clean break there so wholly detached VirtualScroller from the VirtualConsole, which I prefer anyway.